torch/csrc/utils/python_arg_parser.cpp

```diff
@@ -481,6 +481,7 @@ FunctionSignature::FunctionSignature(const std::string& fmt, int index)
       throw std::runtime_error("missing closing parenthesis: " + fmt);
     }
     if (offset == last_offset) {
```

Comment:
This case seems weird. When could this reasonably occur? It seems like, if anything, this code should assert that offset != last_offset.

Reply:
That is, this case only happens (from my reading) if last_offset points to a substring starting with ", ". That is a real possibility if the string is "(, ", for example. But it seems like such strings should be an error?

Reply:
that's one case.  The other case is the `if (offset == std::string::npos) {` case, which would happen with an empty parameter list (last_offset will be equal to offset because last_offset is one past `(` and offset is `)`.

So, I think an assert would work if we fixed up the rest of the logic in some way, I was just a little hesitant to do that because I don't know why it was written this way in the first place...

Reply:
Good point. On rereading I think in the case that `offset == std::string::npos` the code can assert that the`find(')'` returns a position in the string, and then you can check for `offset == last_offset` and short-circuit there. This should avoid the confusion strings like "(, ". I was also worried that setting last_offset like this would result in an invalid start of a substring pointer, but I checked and it's OK to take a substring starting from the null terminator. So the unchecked substrings starting from last_offset should be OK later.